File: python/Interview/S104.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def levelTree(root, last_level):
            if not root:
                return last_level
            l = last_level + 1
            r = last_level + 1
            if root.left:
                l = levelTree(root.left, last_level + 1)
            if root.right:
                r = levelTree(root.right, last_level + 1)
            return max(l, r)

        return levelTree(root, 0)

File: javascript/S104.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    function _depth(root, level, max_level) {
        if(root) {
            max_level.val = Math.max(max_level.val, level + 1);
            _depth(root.left, level + 1, max_level);
            _depth(root.right, level + 1, max_level);
        }
    }
    let max_level = {val: 0};
    _depth(root, 0, max_level);
    return max_level.val;
};